### PR TITLE
Add Python 3.9 builds

### PR DESCRIPTION
**Pull request by chouhanaryan**
Add Python 3.9 builds to the CI.

### CODE REVIEW

**Comment by dstansby** [comment]
Thanks for this! Looks like we might have to wait for https://github.com/actions/virtual-environments/issues/1740 to be resolved so that python 3.9 is available on azure

**Comment by chouhanaryan** [comment]
Yep, I was just about to add a comment linking to it 😅

**Comment by chouhanaryan** [comment]
I'll keep track of that issue and update the PR once it's resolved on their end.

**Comment by Cadair** [comment]
We could for the moment use a docker based build if we are feeling keen: https://openastronomy-azure-pipelines.readthedocs.io/en/latest/run_tox_env.html#other-docker-images

**Comment by nabobalis** [comment]
> We could for the moment use a docker based build if we are feeling keen: [openastronomy-azure-pipelines.readthedocs.io/en/latest/run_tox_env.html#other-docker-images](https://openastronomy-azure-pipelines.readthedocs.io/en/latest/run_tox_env.html#other-docker-images)

Is there really a point to this?

**Comment by Cadair** [comment]
probs not

**Comment by wtbarnes** [comment]
Are we sure that all of our dependencies are 3.9 compatible? I guess there's only one way to find out 😅

**Comment by Cadair** [comment]
3.9 builds are live, but I don't think skimage will build without some hackery at the moment.

**Comment by chouhanaryan** [comment]
Opening and closing to trigger a build. Let's see what happens.

**Comment by Cadair** [comment]
ha lol @ that fail. Can you try the 3.9 build not on macos? also ping @ayshih for macos weirdness.

**Comment by ayshih** [comment]
Not weirdness per se, but the NumPy check is successfully detecting that Accelerate is being linked rather than a non-buggy BLAS/LAPACK library.  From reading the log, it looks it's not the NumPy wheel that's complaining, but rather it's a bundled version of NumPy within Matplotlib 3.3.2 that's complaining?  (Is Matplotlib being built from source here?)  You could try pinning to an earlier version of Matplotlib.

**Comment by Cadair** [comment]
it's probably building numpy from source, as I don't think there are wheels for 3.9 yet.

File: azure-pipelines.yml [hunk 1]
```diff
@@ -50,8 +50,8 @@ jobs:
         - libpng-devel
 
     envs:
-      - macos: py38
-        name: py38_test
+      - macos: py39
```
Comment (Cadair) [inline azure-pipelines.yml:53]:
```suggestion
      - linux: py39
```

**Comment by ayshih** [comment]
Ah, I get the log now.  It built NumPy from source, but it's not until Matplotlib is being built from source that NumPy is imported and hence runs that check for the first time.  For NumPy <1.20, you'd need to explicitly block the linking of the Accelerate libraries on MacOS by setting an environment variable when building NumPy (see https://numpy.org/doc/stable/user/building.html#disabling-atlas-and-other-accelerated-libraries).

**Review by nabobalis — APPROVED** [review]

**Review by kakirastern — APPROVED** [review]

**Comment by nabobalis** [comment]
Thanks again @chouhanaryan

**Pull request merged**